### PR TITLE
Renaming utils build package and function to better suit build and sign

### PR DESCRIPTION
**Pull request by yevgeny-shnaidman**
Utils build function is actualy deals with OCP build entities, and not with KMM builds. This package should also suit the KMM sign flow. This PR renames the package to ocpbuild and changes its functions' names to include OCP

### CODE REVIEW

File: internal/build/ocpbuild/maker.go [hunk 1]
```diff
@@ -138,8 +138,8 @@ func (m *maker) MakeBuildTemplate(
 		ObjectMeta: metav1.ObjectMeta{
 			GenerateName: mld.Name + "-build-",
 			Namespace:    mld.Namespace,
-			Labels:       buildutils.GetBuildLabels(mld, BuildType),
-			Annotations:  buildutils.GetBuildAnnotations(sourceConfigHash),
+			Labels:       ocpbuildutils.GetOCPBuildLabels(mld, BuildType),
+			Annotations:  ocpbuildutils.GetOCPBuildAnnotations(sourceConfigHash),
```
Comment (ybettan) [inline internal/build/ocpbuild/maker.go:142]: Do we also need to add `OCP` to the function names based on the fact there are in the `ocpbuildutils` package?

File: internal/build/ocpbuild/manager.go [hunk 2]
```diff
@@ -43,15 +43,15 @@ func NewManager(
 }
 
 func (m *manager) GarbageCollect(ctx context.Context, modName, namespace string, owner metav1.Object) ([]string, error) {
-	moduleBuilds, err := m.ocpBuildsHelper.GetModuleBuilds(ctx, modName, namespace)
+	moduleBuilds, err := m.ocpBuildsHelper.GetModuleOCPBuilds(ctx, modName, namespace)
```
Comment (ybettan) [inline internal/build/ocpbuild/manager.go:46]: Same.

File: internal/build/ocpbuild/manager.go [hunk 3]
```diff
@@ -43,15 +43,15 @@ func NewManager(
...
 	if err != nil {
 		return nil, fmt.Errorf("failed to get OCP buils for module %s: %v", modName, err)
 	}
 
 	deleteNames := make([]string, 0, len(moduleBuilds))
 	for _, moduleBuild := range moduleBuilds {
 		if moduleBuild.Status.Phase == buildv1.BuildPhaseComplete {
-			err = m.ocpBuildsHelper.DeleteBuild(ctx, &moduleBuild)
+			err = m.ocpBuildsHelper.DeleteOCPBuild(ctx, &moduleBuild)
```
Comment (ybettan) [inline internal/build/ocpbuild/manager.go:54]: Same.

File: internal/utils/ocpbuild/utils.go [hunk 4]
```diff
@@ -12,11 +12,11 @@ const (
 	HashAnnotation = "kmm.node.kubernetes.io/last-hash"
 )
 
-func GetBuildAnnotations(hash uint64) map[string]string {
+func GetOCPBuildAnnotations(hash uint64) map[string]string {
```
Comment (ybettan) [inline internal/utils/ocpbuild/utils.go:15]: Do we also need to add OCP to the function names based on the fact there are in the ocpbuildutils package?
Same question for all other functions in this file.

File: internal/build/ocpbuild/manager.go [hunk 5]
```diff
@@ -22,15 +22,15 @@ import (
...
+	ocpBuildsHelper ocpbuildutils.OpenShiftBuildsHelper
 	authFactory     auth.RegistryAuthGetterFactory
 	registry        registry.Registry
 }
 
 func NewManager(
 	client client.Client,
 	maker Maker,
-	ocpBuildsHelper buildutils.OpenShiftBuildsHelper,
+	ocpBuildsHelper ocpbuildutils.OpenShiftBuildsHelper,
```
Comment (ybettan) [inline internal/build/ocpbuild/manager.go:33]: Should we remove the `Openshift` prefix?

File: internal/sign/ocpbuild/manager.go [hunk 6]
```diff
@@ -22,15 +22,15 @@ import (
...
+	ocpBuildsHelper ocpbuildutils.OpenShiftBuildsHelper
 	authFactory     auth.RegistryAuthGetterFactory
 	registry        registry.Registry
 }
 
 func NewManager(
 	client client.Client,
 	maker Maker,
-	ocpBuildsHelper buildutils.OpenShiftBuildsHelper,
+	ocpBuildsHelper ocpbuildutils.OpenShiftBuildsHelper,
```
Comment (ybettan) [inline internal/sign/ocpbuild/manager.go:33]: Do we need the `Openshift` prefix now that we have `ocp...` in the package name?

File: internal/utils/ocpbuild/helper.go [hunk 7]
```diff
@@ -13,12 +13,12 @@ import (
...
-//go:generate mockgen -source=helper.go -package=build -destination=mock_helper.go
+//go:generate mockgen -source=helper.go -package=ocpbuild -destination=mock_helper.go
 
 type OpenShiftBuildsHelper interface {
-	GetModuleBuildByKernel(ctx context.Context, mld *api.ModuleLoaderData) (*buildv1.Build, error)
-	GetModuleBuilds(ctx context.Context, moduleName, moduleNamespace string) ([]buildv1.Build, error)
-	DeleteBuild(ctx context.Context, build *buildv1.Build) error
+	GetModuleOCPBuildByKernel(ctx context.Context, mld *api.ModuleLoaderData) (*buildv1.Build, error)
+	GetModuleOCPBuilds(ctx context.Context, moduleName, moduleNamespace string) ([]buildv1.Build, error)
+	DeleteOCPBuild(ctx context.Context, build *buildv1.Build) error
```
Comment (ybettan) [inline internal/utils/ocpbuild/helper.go:21]: Should we remove the `OpenShift` prefix from the object? How about removing `OCP` from the methods?
We can always name the filed holding that object `ocpBuildHelper`. Do we also want it in the object/methods names?

File: internal/utils/ocpbuild/utils.go (hunk 4 above)
Comment (yevgeny-shnaidman) [reply internal/utils/ocpbuild/utils.go:15]: I would change it to OCP and keep it, makes code clearer

File: internal/sign/ocpbuild/manager.go (hunk 6 above)
Comment (yevgeny-shnaidman) [reply internal/sign/ocpbuild/manager.go:33]: same answer as above

File: internal/utils/ocpbuild/helper.go (hunk 7 above)
Comment (yevgeny-shnaidman) [reply internal/utils/ocpbuild/helper.go:21]: same answer as above

File: internal/sign/ocpbuild/manager.go (hunk 6 above)
Comment (ybettan) [reply internal/sign/ocpbuild/manager.go:33]: Then how about we change `Openshift` to `OCP` to be consistent?

Comment (yevgeny-shnaidman) [reply internal/sign/ocpbuild/manager.go]: fixed

**Pull request merged**